modules/3d/src/rgbd/odometry_functions.hpp

```diff
@@ -100,7 +117,23 @@ typedef
 void (*CalcRgbdEquationCoeffsPtr)(double*, double, double, const Point3f&, double, double);
 
 static inline
-void calcICPEquationCoeffs(double* C, const Point3f& p0, const Vec3f& n1)
+void calcICPScaleEquationCoeffs(double* C, const Point3f& p0, const Point3f& p1, const Vec3f& n1)
+{
+    C[0] = -p0.z * n1[1] + p0.y * n1[2];
+    C[1] = p0.z * n1[0] - p0.x * n1[2];
+    C[2] = -p0.y * n1[0] + p0.x * n1[1];
+    C[3] = n1[0];
+    C[4] = n1[1];
+    C[5] = n1[2];
+    double diff = n1.dot(p1 - p0);
+    if (diff < DBL_EPSILON || abs(diff) > 1000000)
+        C[6] = 0;
+    else
+        C[6] = n1.dot(p1-p0);
```

Comment:
This should be checked if this formula is to be fixed or not:
it's possible that there should be `n1.dot((p0 - currentTranslation)/currentScale)`, it's easier to do at caller's side.